restore auto-derivation for show

kailuowang opened the pull request:
@joroKr21 mind to take a look? 
Once this is approved I'll add more. 
I also namespaced the implicit methods to avoid potential conflicts.

joroKr21 commented on core/src/main/scala/cats/derived/show.scala:
```diff
@@ -21,23 +21,16 @@ object MkShow extends MkShowDerivation {
   implicit def productDerivedShowFurther[K <: Symbol, V, T <: HList](
        implicit key: Witness.Aux[K],
-       showV: Lazy[MkShow[V]],
+       showV: Show[V] OrElse MkShow[V],
```
I think we should do this for coproducts as well. Although less likely that someone would define a special case only for one subclass, it's still a possibility.

kailuowang replied:
thanks I missed that.

milessabin commented:
Looks good, but there are a couple of renamings that appear to be unrelated to the essence of the PR: `mkEmpty` -> `kittensMkEmpty` and `mkFunctor` -> `kittensMkFunctor`. Why are they included here?

kailuowang commented:
@milessabin they are to avoid implicit conflicts. In this PR, I introduced a new one of the same kind, namely `kittenMkShow`. To make it consistent I renamed the other two and hopefully setup a convention for naming schema. We can certainly do that in a separate PR if you prefer.

milessabin commented:
What do they conflict with?

kailuowang commented:
Nothing in particular at the moment, but if some other lib for some reason also has an implicit method called `mkShow` which isn't very inconceivable, it will conflict with our import. This is to avoid the potential implicit name conflict when both libs are imported.

milessabin commented:
Gotcha. OK, no objections ... LGTM :-)

kailuowang commented:
thanks!
